src/ambulance_game/markov/waiting.py

```diff
@@ -303,56 +297,63 @@ def mean_waiting_time_formula(
     """
-    if formula == "recursive":
-        mean_waiting_time = mean_waiting_time_formula_using_recursive_approach(
-            all_states=all_states,
-            pi=pi,
-            class_type=class_type,
-            lambda_2=lambda_2,
-            lambda_1=lambda_1,
-            mu=mu,
-            num_of_servers=num_of_servers,
-            threshold=threshold,
-            system_capacity=system_capacity,
-            buffer_capacity=buffer_capacity,
-        )
-
-    if formula == "algebraic":
-        mean_waiting_time = mean_waiting_time_formula_using_algebraic_approach(
-            all_states=all_states,
-            pi=pi,
-            class_type=class_type,
-            lambda_2=lambda_2,
-            lambda_1=lambda_1,
-            mu=mu,
-            num_of_servers=num_of_servers,
-            threshold=threshold,
-            system_capacity=system_capacity,
-            buffer_capacity=buffer_capacity,
-        )
+    mean_waiting_time_class_1 = waiting_formula(
+        all_states=all_states,
+        pi=pi,
+        class_type=1,
+        lambda_2=lambda_2,
+        lambda_1=lambda_1,
+        mu=mu,
+        num_of_servers=num_of_servers,
+        threshold=threshold,
+        system_capacity=system_capacity,
+        buffer_capacity=buffer_capacity,
+    )
+    mean_waiting_time_class_2 = waiting_formula(
+        all_states=all_states,
+        pi=pi,
+        class_type=2,
+        lambda_2=lambda_2,
+        lambda_1=lambda_1,
+        mu=mu,
+        num_of_servers=num_of_servers,
+        threshold=threshold,
+        system_capacity=system_capacity,
+        buffer_capacity=buffer_capacity,
+    )
 
-    if formula == "closed_form":
-        mean_waiting_time = mean_waiting_time_formula_using_closed_form_approach(
-            all_states=all_states,
-            pi=pi,
-            class_type=class_type,
-            lambda_2=lambda_2,
-            lambda_1=lambda_1,
-            mu=mu,
-            num_of_servers=num_of_servers,
-            threshold=threshold,
-            system_capacity=system_capacity,
-            buffer_capacity=buffer_capacity,
-        )
+    prob_accept_class_1 = np.sum(
+        [
+            pi[state]
+            for state in all_states
+            if is_accepting_state(state, 1, threshold, system_capacity, buffer_capacity)
+        ]
+    )
+    prob_accept_class_2 = np.sum(
+        [
+            pi[state]
+            for state in all_states
+            if is_accepting_state(state, 2, threshold, system_capacity, buffer_capacity)
+        ]
+    )
+    class_2_rate = (lambda_2 * prob_accept_class_2) / (
+        (lambda_2 * prob_accept_class_2) + (lambda_1 * prob_accept_class_1)
+    )
+    class_1_rate = (lambda_1 * prob_accept_class_1) / (
+        (lambda_2 * prob_accept_class_2) + (lambda_1 * prob_accept_class_1)
+    )
 
-    return mean_waiting_time
+    return (
+        mean_waiting_time_class_2 * class_2_rate
+        + mean_waiting_time_class_1 * class_1_rate
```

Comment:
All of this will be relatively simplified by just having two tuples instead of actual named vars.